parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java

```diff
@@ -273,11 +261,45 @@ private ThriftType convertClassToThriftType(Class typeClass) {
       return new ThriftType.I64Type();
     } else if (typeClass == String.class) {
       return new ThriftType.StringType();
+    } else if (typeClass == scala.collection.Seq.class){
+      Manifest<?> a = typeManifest.typeArguments().apply(0);
+      return convertListTypeField(name, a, requirement);
+    } else if (typeClass == scala.collection.Set.class){
+      Manifest<?> setElementManifest = typeManifest.typeArguments().apply(0);
+      return convertSetTypeField(name, setElementManifest, requirement);
+    } else if (typeClass == scala.collection.Map.class){
+      List<Manifest<?>> ms = JavaConversions.seqAsJavaList(typeManifest.typeArguments());
+      Manifest keyManifest = ms.get(0);
+      Manifest valueManifest = ms.get(1);
+      return convertMapTypeField(name, keyManifest, valueManifest, requirement);
+    } else if (com.twitter.scrooge.ThriftEnum.class.isAssignableFrom(typeClass)){
+      return convertEnumTypeField(typeClass, name);
     } else {
       return convertStructFromClass(typeClass);
     }
   }
 
+  /**
+   * There is a bug/inconsistency in ThriftStructConverter:
+   * The field name of a nested list/set has the same name as its parent. for example:
+   *  1: required list<list<int>> lm
+   * the name of the inner list is lm, but is should be lm_list_elem.
+   *
+   * This bug/inconsistency does not affect the converted parquet schema because ThriftSchemaConverter ignores the name
+   * of inner list.
+   *
+   * Here in ScroogeStructConverter the same behavior is replicated to generate the same ThriftStruct with
+   * ThriftStructConverter
+   *
+   * TODO: This fix might not be necessary since parquet mostly operates on MessageType, the ThriftStruct is an intermediate
+   * layer for generating MessageType from a thrift/scrooge class. The fix could be removed once we are confident it's
+   * safe for doing so.
+   * @return
+   */
+  private String fixNestListOrSetName(String name) {
+    return name.replaceAll("_list_elem|_set_elem", "");
```

Comment:
This bug might be trickier than you thought... ThriftSchema converter use the name that is converted by elephant_bird. but for the list element, it's using the container's name. But for map, it's using the correct key name. And the tricky case comes here: when you have list<list<list<map>>>  the name of map key is ls_list_elem_list_elem_list_elem_map_key, but the parent's name is ls. Basically all the parents do not have the _list_elem suffixies due to the bug in TBase implementation, but the leaf node should have the _list_elem suffixes.  I have unit tests for this which you can reference.  As I said, this fix is purely for being conservative and consistent with TBase implementation. Hopefully I can do a more thorough test and remove this fix/hack